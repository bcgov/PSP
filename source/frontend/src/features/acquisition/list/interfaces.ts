export interface IAcquisitionFilter {
  acquisitionFileStatusTypeCode: string;
  acquisitionFileNameOrNumber: string;
  projectNameOrNumber: string;
  searchBy: string;
  pin: string;
  pid: string;
  address: string;
}
