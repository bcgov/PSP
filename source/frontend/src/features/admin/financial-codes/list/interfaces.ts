export interface IFinancialCodesFilter {
  acquisitionFileStatusTypeCode: string;
  acquisitionFileNameOrNumber: string;
  projectNameOrNumber: string;
  searchBy: string;
  pin: string;
  pid: string;
  address: string;
}
