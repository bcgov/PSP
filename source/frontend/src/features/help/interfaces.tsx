export interface IHelpForm {
  user: string;
  email: string;
  page: string;
}
