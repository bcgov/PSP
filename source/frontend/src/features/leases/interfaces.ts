export interface ILeaseFilter {
  pid: string;
  pin: string;
  lFileNo: string;
  searchBy: string;
  leaseStatusTypes: string[];
  tenantName: string;
  programs: string[];
  expiryStartDate: string;
  expiryEndDate: string;
  regionType: string;
  details: string;
}

export interface ILeaseSearchBy {
  pin: string;
  pid: string;
  address: string;
  lFileNo: string;
  historical: string;
}
