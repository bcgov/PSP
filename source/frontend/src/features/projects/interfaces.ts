export interface IProjectFilter {
  projectRegionCode: string;
  projectStatusCode: string;
  projectName: string;
  projectNumber: string;
}

export interface IProjectSearchBy {
  projectNumber: string;
  projectName: string;
}
