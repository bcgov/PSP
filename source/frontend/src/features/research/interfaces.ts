export interface IResearchFilter {
  regionCode: string;
  researchFileStatusTypeCode: string;
  name: string;
  roadOrAlias: string;
  rfileNumber: string;
  createOrUpdateRange: string;
  createOrUpdateBy: string;
  researchSearchBy: string;
  appCreateUserid: string;
  createdOnStartDate: string;
  createdOnEndDate: string;
  appLastUpdateUserid: string;
  updatedOnStartDate: string;
  updatedOnEndDate: string;
}
