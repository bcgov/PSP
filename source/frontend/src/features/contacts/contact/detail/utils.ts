import { ContactInfoField } from 'features/contacts/interfaces';
import { Dictionary } from 'interfaces/Dictionary';
import { IContactAddress, IContactMethod } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export function getContactInfo(
  validTypes: Dictionary<string>,
  contactMethods?: IContactMethod[],
): ContactInfoField[] {
  if (contactMethods === undefined) {
    return [];
  }
  // Get only the valid types
  let filteredFields = contactMethods.reduce(
    (accumulator: ContactInfoField[], method: IContactMethod) => {
      if (Object.keys(validTypes).includes(method.contactMethodType.id)) {
        accumulator.push({
          info: method.value,
          label: validTypes[method.contactMethodType.id],
        });
      }
      return accumulator;
    },
    [],
  );

  // Sort according to the dictionary order
  return filteredFields.sort((a, b) => {
    return Object.values(validTypes).indexOf(a.label) - Object.values(validTypes).indexOf(b.label);
  });
}

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
