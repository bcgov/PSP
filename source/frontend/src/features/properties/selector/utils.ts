import { IMapProperty } from 'features/properties/selector/models';
import { compact } from 'lodash';
import { Api_PropertyFile } from 'models/api/PropertyFile';
import { formatApiAddress, pidFormatter } from 'utils';

import { Api_Geometry, Api_Property } from './../../../models/api/Property';

export enum NameSourceType {
  PID = 'PID',
  PIN = 'PIN',
  PLAN = 'Plan #',
  LOCATION = 'Location',
  NAME = 'Descriptive Name',
  NONE = 'n/a',
  ADDRESS = 'Address',
}

interface PropertyName {
  label: NameSourceType;
  value: string;
}

export const getPropertyName = (property: IMapProperty): PropertyName => {
  if (property.pid !== undefined && property.pid?.toString().length > 0 && property.pid !== '0') {
    return { label: NameSourceType.PID, value: pidFormatter(property.pid.toString()) };
  } else if (
    property.pin !== undefined &&
    property.pin?.toString()?.length > 0 &&
    property.pin !== '0'
  ) {
    return { label: NameSourceType.PIN, value: property.pin.toString() };
  } else if (property.planNumber !== undefined) {
    return { label: NameSourceType.PLAN, value: property.planNumber };
  } else if (property.latitude !== undefined && property.longitude !== undefined) {
    return {
      label: NameSourceType.LOCATION,
      value: compact([property.longitude?.toFixed(6), property.latitude?.toFixed(6)]).join(', '),
    };
  } else if (property.address !== undefined) {
    return {
      label: NameSourceType.ADDRESS,
      value: property.address,
    };
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getPrettyLatLng = (location?: Api_Geometry) =>
  compact([
    location?.coordinate?.x?.toFixed(6) ?? 0,
    location?.coordinate?.y?.toFixed(6) ?? 0,
  ]).join(', ');

export const getFilePropertyName = (
  fileProperty?: Api_PropertyFile,
  skipName: boolean = false,
): PropertyName => {
  if (fileProperty === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  if (
    fileProperty.propertyName !== undefined &&
    fileProperty.propertyName !== '' &&
    skipName === false
  ) {
    return { label: NameSourceType.NAME, value: fileProperty.propertyName };
  } else if (fileProperty.property !== undefined) {
    const property = fileProperty.property;
    return getApiPropertyName(property);
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getApiPropertyName = (property?: Api_Property): PropertyName => {
  if (property === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  let mapProperty: IMapProperty = {
    pin: property.pin?.toString(),
    pid: property.pid?.toString(),
    latitude: property.latitude,
    longitude: property.longitude,
    planNumber: property.planNumber,
    address: property.address !== undefined ? formatApiAddress(property.address) : undefined,
  };
  return getPropertyName(mapProperty);
};
