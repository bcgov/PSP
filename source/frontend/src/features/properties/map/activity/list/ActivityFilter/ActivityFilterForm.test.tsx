import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { Formik } from 'formik';
import { createMemoryHistory } from 'history';
import { defaultActivityFilter } from 'interfaces/IActivityResults';
import { noop } from 'lodash';
import { mockLookups } from 'mocks';
import { lookupCodesSlice } from 'store/slices/lookupCodes';
import { fillInput, renderAsync, RenderOptions } from 'utils/test-utils';

import { ActivityFilterForm, IActivityFilterFormProps } from './ActivityFilterForm';

const history = createMemoryHistory();
const mockAxios = new MockAdapter(axios);
const onSetFilter = jest.fn();
const storeState = {
  [lookupCodesSlice.name]: { lookupCodes: mockLookups },
};

describe('ActivityFilterForm component', () => {
  const setup = async (
    renderOptions: RenderOptions &
      Partial<IActivityFilterFormProps> & {
        initialValues?: any;
      } = {},
  ) => {
    // render component under test
    const component = await renderAsync(
      <Formik initialValues={renderOptions.initialValues ?? {}} onSubmit={noop}>
        <ActivityFilterForm onSetFilter={onSetFilter} activityFilter={defaultActivityFilter} />
      </Formik>,
      {
        ...renderOptions,
        store: storeState,
        history,
      },
    );

    return {
      component,
    };
  };

  beforeEach(() => {
    mockAxios.reset();
  });

  it('renders as expected', async () => {
    const { component } = await setup({});

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders with data as expected', async () => {
    const { component } = await setup({
      initialValues: { ...defaultActivityFilter },
    });

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('renders activityType with data as expected', async () => {
    const {
      component: { getByTestId, container },
    } = await setup({
      initialValues: { ...defaultActivityFilter },
    });

    await fillInput(container, 'activityType', 'Survey', 'select');

    expect(getByTestId('activity-type')).not.toBeNull();
  });

  it('renders activityStatus with data as expected', async () => {
    const {
      component: { getByTestId, container },
    } = await setup({
      initialValues: { ...defaultActivityFilter },
    });

    await fillInput(container, 'activityStatus', 'Draft', 'select');

    expect(getByTestId('activity-status')).not.toBeNull();
  });
});
