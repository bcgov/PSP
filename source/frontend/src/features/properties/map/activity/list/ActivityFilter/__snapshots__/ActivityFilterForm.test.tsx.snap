// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityFilterForm component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="container"
  />
  .c1.btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.4rem 1.2rem;
  min-height: 3rem;
  border: 0.2rem solid transparent;
  border-radius: 0.4rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  font-family: 'BCSans','Noto Sans',Verdana,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  cursor: pointer;
}

.c1.btn:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c1.btn:focus {
  outline-width: 0.4rem;
  outline-style: solid;
  outline-offset: 1px;
  box-shadow: none;
}

.c1.btn.btn-primary {
  border: none;
}

.c1.btn.btn-secondary {
  background: none;
}

.c1.btn.btn-info {
  border: none;
  background: none;
  padding-left: 0.6rem;
  padding-right: 0.6rem;
}

.c1.btn.btn-info:hover,
.c1.btn.btn-info:active,
.c1.btn.btn-info:focus {
  background: none;
}

.c1.btn.btn-light {
  border: none;
}

.c1.btn.btn-dark {
  border: none;
}

.c1.btn.btn-link {
  font-size: 1.6rem;
  font-weight: 400;
  background: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 2.5rem;
  line-height: 3rem;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  text-align: left;
  padding: 0;
}

.c1.btn.btn-link:hover,
.c1.btn.btn-link:active,
.c1.btn.btn-link:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  border: none;
  background: none;
  box-shadow: none;
  outline: none;
}

.c1.btn.btn-link:disabled,
.c1.btn.btn-link.disabled {
  background: none;
  pointer-events: none;
}

.c1.btn:disabled,
.c1.btn:disabled:hover {
  box-shadow: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.65;
}

.c1.Button .Button__icon {
  margin-right: 1.6rem;
}

.c1.Button--icon-only:focus {
  outline: none;
}

.c1.Button--icon-only .Button__icon {
  margin-right: 0;
}

.c0 {
  border-radius: 0.5rem;
}

<form
    class="c0 p-3"
    novalidate=""
  >
    <div
      class="row"
    >
      <div
        class="col-auto"
      >
        <label>
          Filter by:
        </label>
      </div>
      <div
        class="col-4"
      >
        <div
          class="form-group"
        >
          <select
            class="form-select form-control"
            data-testid="activity-type"
            id="input-activityTypeId"
            name="activityTypeId"
          >
            <option
              value=""
            >
              All activity types
            </option>
            <option
              class="option"
              value="GENERAL"
            >
              General
            </option>
            <option
              class="option"
              value="GENLTR"
            >
              Generate Letter
            </option>
            <option
              class="option"
              value="SITEVIS"
            >
              Site Visit
            </option>
            <option
              class="option"
              value="SURVEY"
            >
              Survey
            </option>
          </select>
        </div>
      </div>
      <div
        class="col-4"
      >
        <div
          class="form-group"
        >
          <select
            class="form-select form-control"
            data-testid="activity-status"
            id="input-status"
            name="status"
          >
            <option
              value=""
            >
              All statuses
            </option>
            <option
              class="option"
              value="NOSTART"
            >
              Not Started
            </option>
            <option
              class="option"
              value="INPROG"
            >
              In Progress
            </option>
            <option
              class="option"
              value="COMPLETE"
            >
              Complete
            </option>
            <option
              class="option"
              value="CANCELLED"
            >
              Cancelled
            </option>
          </select>
        </div>
      </div>
      <div
        class="col-1"
      >
        <button
          class="c1 Button Button--icon-only primary btn btn-primary"
          data-testid="search"
          id="search-button"
          title="search"
          type="submit"
        >
          <div
            class="Button__icon"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        class="col-1"
      >
        <button
          class="c1 Button Button--icon-only btn btn-info"
          data-testid="reset-button"
          id="reset-button"
          title="reset-button"
          type="reset"
        >
          <div
            class="Button__icon"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M212.333 224.333H12c-6.627 0-12-5.373-12-12V12C0 5.373 5.373 0 12 0h48c6.627 0 12 5.373 12 12v78.112C117.773 39.279 184.26 7.47 258.175 8.007c136.906.994 246.448 111.623 246.157 248.532C504.041 393.258 393.12 504 256.333 504c-64.089 0-122.496-24.313-166.51-64.215-5.099-4.622-5.334-12.554-.467-17.42l33.967-33.967c4.474-4.474 11.662-4.717 16.401-.525C170.76 415.336 211.58 432 256.333 432c97.268 0 176-78.716 176-176 0-97.267-78.716-176-176-176-58.496 0-110.28 28.476-142.274 72.333h98.274c6.627 0 12 5.373 12 12v48c0 6.627-5.373 12-12 12z"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`ActivityFilterForm component renders with data as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="container"
  />
  .c1.btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.4rem 1.2rem;
  min-height: 3rem;
  border: 0.2rem solid transparent;
  border-radius: 0.4rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  font-family: 'BCSans','Noto Sans',Verdana,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  cursor: pointer;
}

.c1.btn:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c1.btn:focus {
  outline-width: 0.4rem;
  outline-style: solid;
  outline-offset: 1px;
  box-shadow: none;
}

.c1.btn.btn-primary {
  border: none;
}

.c1.btn.btn-secondary {
  background: none;
}

.c1.btn.btn-info {
  border: none;
  background: none;
  padding-left: 0.6rem;
  padding-right: 0.6rem;
}

.c1.btn.btn-info:hover,
.c1.btn.btn-info:active,
.c1.btn.btn-info:focus {
  background: none;
}

.c1.btn.btn-light {
  border: none;
}

.c1.btn.btn-dark {
  border: none;
}

.c1.btn.btn-link {
  font-size: 1.6rem;
  font-weight: 400;
  background: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 2.5rem;
  line-height: 3rem;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  text-align: left;
  padding: 0;
}

.c1.btn.btn-link:hover,
.c1.btn.btn-link:active,
.c1.btn.btn-link:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  border: none;
  background: none;
  box-shadow: none;
  outline: none;
}

.c1.btn.btn-link:disabled,
.c1.btn.btn-link.disabled {
  background: none;
  pointer-events: none;
}

.c1.btn:disabled,
.c1.btn:disabled:hover {
  box-shadow: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.65;
}

.c1.Button .Button__icon {
  margin-right: 1.6rem;
}

.c1.Button--icon-only:focus {
  outline: none;
}

.c1.Button--icon-only .Button__icon {
  margin-right: 0;
}

.c0 {
  border-radius: 0.5rem;
}

<form
    class="c0 p-3"
    novalidate=""
  >
    <div
      class="row"
    >
      <div
        class="col-auto"
      >
        <label>
          Filter by:
        </label>
      </div>
      <div
        class="col-4"
      >
        <div
          class="form-group"
        >
          <select
            class="form-select form-control"
            data-testid="activity-type"
            id="input-activityTypeId"
            name="activityTypeId"
          >
            <option
              value=""
            >
              All activity types
            </option>
            <option
              class="option"
              value="GENERAL"
            >
              General
            </option>
            <option
              class="option"
              value="GENLTR"
            >
              Generate Letter
            </option>
            <option
              class="option"
              value="SITEVIS"
            >
              Site Visit
            </option>
            <option
              class="option"
              value="SURVEY"
            >
              Survey
            </option>
          </select>
        </div>
      </div>
      <div
        class="col-4"
      >
        <div
          class="form-group"
        >
          <select
            class="form-select form-control"
            data-testid="activity-status"
            id="input-status"
            name="status"
          >
            <option
              value=""
            >
              All statuses
            </option>
            <option
              class="option"
              value="NOSTART"
            >
              Not Started
            </option>
            <option
              class="option"
              value="INPROG"
            >
              In Progress
            </option>
            <option
              class="option"
              value="COMPLETE"
            >
              Complete
            </option>
            <option
              class="option"
              value="CANCELLED"
            >
              Cancelled
            </option>
          </select>
        </div>
      </div>
      <div
        class="col-1"
      >
        <button
          class="c1 Button Button--icon-only primary btn btn-primary"
          data-testid="search"
          id="search-button"
          title="search"
          type="submit"
        >
          <div
            class="Button__icon"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        class="col-1"
      >
        <button
          class="c1 Button Button--icon-only btn btn-info"
          data-testid="reset-button"
          id="reset-button"
          title="reset-button"
          type="reset"
        >
          <div
            class="Button__icon"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M212.333 224.333H12c-6.627 0-12-5.373-12-12V12C0 5.373 5.373 0 12 0h48c6.627 0 12 5.373 12 12v78.112C117.773 39.279 184.26 7.47 258.175 8.007c136.906.994 246.448 111.623 246.157 248.532C504.041 393.258 393.12 504 256.333 504c-64.089 0-122.496-24.313-166.51-64.215-5.099-4.622-5.334-12.554-.467-17.42l33.967-33.967c4.474-4.474 11.662-4.717 16.401-.525C170.76 415.336 211.58 432 256.333 432c97.268 0 176-78.716 176-176 0-97.267-78.716-176-176-176-58.496 0-110.28 28.476-142.274 72.333h98.274c6.627 0 12 5.373 12 12v48c0 6.627-5.373 12-12 12z"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
